python/paddle/fluid/layers/nn.py

```diff
@@ -2797,6 +2797,10 @@ def batch_norm(input,
     helper = LayerHelper('batch_norm', **locals())
     dtype = helper.input_dtype()
 
+    # use fp32 for bn parameter
+    if dtype == core.VarDesc.VarType.FP16:
+        dtype = core.VarDesc.VarType.FP32
```

Comment:
what's the output type, is it fp16 or fp32. What if the next op needs fp16?

Reply:
this is strange. why input is fp16, output is fp32? What if I want to infer type, output should by fp16 when input is fp16?

Reply:
The output is fp16.

explicit is better than implicit, should add an argument to control this. Will do this.